### PR TITLE
Fix attributes canceling each other

### LINKED ISSUE #353: `rename` attributes overwrite each other
```rust
#![feature(custom_derive, plugin)]
#![plugin(serde_macros)]

extern crate serde_json;

#[derive(Serialize, Deserialize, Default)]
struct S {
    #[serde(rename(serialize="s1", deserialize="d1"))]
    v1: (),
    #[serde(rename(serialize="s2"))]
    #[serde(rename(deserialize="d2"))]
    v2: (),
}

fn main() {
    let s: S = Default::default();
    println!("{}", serde_json::to_string(&s).unwrap());
}
```

Expected: `{"s1":null,"s2":null}`
Actual: `{"s1":null,"v2":null}`

### PR DESCRIPTION
**Pull request by dtolnay**
Fixes #353.

### CODE REVIEW

File: serde_codegen/src/attr.rs [hunk 1]
```diff
@@ -82,17 +82,19 @@ impl ContainerAttrs {
...
                     }
 
                     // Parse `#[serde(rename(serialize="foo", deserialize="bar"))]`
                     ast::MetaItemKind::List(ref name, ref meta_items) if name == &"rename" => {
                         let (ser_name, de_name) = try!(get_renames(cx, meta_items));
-
-                        container_attrs.name.serialize_name = ser_name;
-                        container_attrs.name.deserialize_name = de_name;
+                        if ser_name.is_some() {
+                            container_attrs.name.serialize_name = ser_name;
```
Comment (oli-obk) [inline serde_codegen/src/attr.rs]: I think this should also check whether this is overwriting a previous value and emit an error (or at least a warning) if it is.

**Comment by oli-obk** [comment]
Ok... it really is time for cfail tests... I'm on it ^^

File: serde_codegen/src/attr.rs (hunk 1 above)
Comment (dtolnay) [inline serde_codegen/src/attr.rs]: Good call. Bedtime for me but I filed #355 and I will try to follow up this week.

**Comment by oli-obk** [comment]
@homu r+

**Pull request merged**